src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java

```diff
@@ -0,0 +1,90 @@
+package edu.wpi.grip.core.operations.composite;
+
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.operations.opencv.CVOperation;
+import org.bytedeco.javacpp.opencv_core;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.bytedeco.javacpp.opencv_core.inRange;
+import static org.bytedeco.javacpp.opencv_imgproc.COLOR_BGR2HLS;
+import static org.bytedeco.javacpp.opencv_imgproc.cvtColor;
+
+/**
+ * An {@link edu.wpi.grip.core.Operation} that converts a color image into a binary image based on the HLS threshold ranges for each channel
+ */
+public class HLSThresholdOperation implements CVOperation {
+    private final SocketHint<opencv_core.Mat> inputHint = new SocketHint<opencv_core.Mat>("Input", opencv_core.Mat.class, opencv_core.Mat::new);
+    private final SocketHint<List> hueHint = new SocketHint<List>("Hue", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 180.0)});
+    private final SocketHint<List> luminanceHint = new SocketHint<List>("Luminance", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 255.0)});
+    private final SocketHint<List> saturationHint = new SocketHint<List>("Saturation", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 255.0)});
+
+    private final SocketHint<opencv_core.Mat> outputHint = new SocketHint<opencv_core.Mat>("output", opencv_core.Mat.class, opencv_core.Mat::new);
+
+    @Override
+    public String getName() {
+        return "HLS Threshold";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Segment an image based on hue, luminance and saturation ranges";
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        return new InputSocket<?>[]{
+                new InputSocket<>(eventBus, inputHint),
+                new InputSocket<>(eventBus, hueHint),
```

Comment:
I was following the CV standards. Is this an idea or something you expect to be changed?
I don't care myself.